dapr_agents/memory/base.py

```diff
@@ -73,3 +75,16 @@ def reset_memory(self):
             This method must be implemented by subclasses.
         """
         pass
+
+    @staticmethod
+    def _convert_to_dict(message: Union[Dict, BaseMessage]) -> Dict:
```

Comment:
is there a way we can move away from having the typed BaseMessage and the dict version? Or are both required?

Reply:
That is a good question. I think the BaseMessage is more so what people can do thins like

```python
UserMessage("Hello")
AssistantMessage("Hello, how can I help you?")
```

Those are based on `BaseMessage`.

Reply:
It is just for flexibility. If a user interacts with those methods to add messages, writing JSON objects like `{"role": "user", "content":"hello"}` might be a little bit annoying. Having the `BaseMessage` style described in my previous comment, makes it easy. Therefore, when dealing with memory, I would expect also a user trying to add `BaseMessage` instances directly.